### PR TITLE
codeintel: Convert recordExpirer janitor job to expiredUploadDeleter

### PR DESCRIPTION
**Pull request by efritz**
This PR disables the old janitor job that would soft delete/delete upload and index records (respectively) based only on age and adds a new job that soft deletes uploads based on their reference and expiration status (calculated in the background).

Prerequisites:
- https://github.com/sourcegraph/sourcegraph/pull/24794
- https://github.com/sourcegraph/sourcegraph/pull/24827
- https://github.com/sourcegraph/sourcegraph/pull/24294

### CODE REVIEW

File: enterprise/cmd/worker/internal/codeintel/janitor/record_expirer.go [hunk 1]
```diff
@@ -1,88 +0,0 @@
...
-}
-
-func (e *recordExpirer) expireIndexes(ctx context.Context) error {
-	tx, err := e.dbStore.Transact(ctx)
-	if err != nil {
-		return err
-	}
-	defer func() { err = tx.Done(err) }()
-
-	count, err := tx.DeleteOldIndexes(ctx, e.ttl, time.Now())
```
Comment (eseliger) [inline enterprise/cmd/worker/internal/codeintel/janitor/record_expirer.go:78]: can these store methods go, too?

**Review by eseliger — APPROVED** [review]

File: enterprise/cmd/worker/internal/codeintel/janitor/record_expirer.go (hunk 1 above)
Comment (efritz) [reply enterprise/cmd/worker/internal/codeintel/janitor/record_expirer.go:78]: Yes was going to do a cleanup PR to keep the diff simple.

Comment (efritz) [reply enterprise/cmd/worker/internal/codeintel/janitor/record_expirer.go:78]: Nvm deleting stuff now.

**Comment by efritz** [comment]
Fixes https://github.com/sourcegraph/sourcegraph/issues/24742.

**Pull request merged**